### PR TITLE
Move zwave patch to repo

### LINKED ISSUE #2599: Travis Linux build broken
The travis script fails to patch zwave example build: https://travis-ci.org/domoticz/domoticz/builds/408853253?utm_source=github_status&utm_medium=notification

Build output causing the error:
```
$ (cd open-zwave-read-only; wget http://www.domoticz.com/domo.patch; patch -p1 < domo.patch; make)
--2018-07-27 08:57:32--  http://www.domoticz.com/domo.patch
Resolving www.domoticz.com (www.domoticz.com)... 185.27.142.198
Connecting to www.domoticz.com (www.domoticz.com)|185.27.142.198|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 20280 (20K)
Saving to: ‘domo.patch’
100%[======================================>] 20,280      --.-K/s   in 0.1s    
2018-07-27 08:57:32 (180 KB/s) - ‘domo.patch’ saved [20280/20280]
patching file cpp/build/Makefile
Hunk #1 succeeded at 65 with fuzz 1 (offset 6 lines).
patching file cpp/hidapi/hidapi/hidapi.h
patching file cpp/hidapi/libusb/hid.c
Hunk #2 succeeded at 1430 (offset 6 lines).
patching file cpp/hidapi/linux/hid.c
patching file cpp/hidapi/mac/hid.c
patching file cpp/hidapi/windows/hid.cpp
patching file cpp/src/Driver.cpp
Hunk #2 succeeded at 226 (offset 2 lines).
patching file cpp/src/command_classes/CentralScene.cpp
patching file cpp/src/command_classes/CentralScene.h
patching file cpp/src/platform/HidController.cpp
patching file cpp/src/platform/HidController.h
patching file cpp/src/platform/unix/SerialControllerImpl.cpp
Reversed (or previously applied) patch detected!  Assume -R? [n] 
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
```

### PR DESCRIPTION
**Pull request by emontnemery**
Intended to fix #2599.
@gizmocuz Please have a look, I don't use ZWave so I can't test if the build is good.

### CODE REVIEW

**Comment by emontnemery** [comment]
(Reopened/closed to restart the Travis build)

**Comment by gizmocuz** [comment]
Thanks for this !
Are the udev patches still needed ? If you check out the latest OZW build it should it should be build without udev/HID support by default correct?

**Comment by Jumbotroll** [comment]
Using: Make USE_HID=0 when build openzwave for synology working great. No need for udev patches, but still using the 1125.patch.

**Comment by emontnemery** [comment]
OK, I removed the udev patches

File: .travis.yml [hunk 1]
```diff
@@ -68,7 +68,7 @@ matrix:
       - ./b2 stage threading=multi link=static --with-thread --with-system --with-chrono >> /dev/null 2>&1
       - sudo ./b2 install threading=multi link=static --with-thread --with-system --with-chrono >> /dev/null 2>&1
       - cd ..
-      - (cd open-zwave-read-only; wget http://www.domoticz.com/domo.patch; patch -p1 < domo.patch; make)
+      - (cd open-zwave-read-only; patch -p1 < ../patches/domoticz-open-zwave.patch; make)
```
Comment (gizmocuz) [inline .travis.yml:71]: Not sure if this is going to work.
Someone checks out the git repo in a 'dev-domoticz' folder
you cd to open-zwave-read-only
which is one folder down
../patches does not exist, ../dev-domoticz/patches/... does exists
What is someone checks out domoticz in 'domoticz' or 'domosource' or ....
I think we need to keep the patch in our www folder to easy overcome the problem ?

**Comment by emontnemery** [comment]
@gizmocuz I don't understand your comment.. The Travis build is working fine since open zwave is cloned to within the domoticz tree by the script. Are you talking about how to clearly document the use of the patch when building manually??

**Comment by gizmocuz** [comment]
If i look at it, the domoticz code is checked out in:
~/dev-domoticz

openzwave is checked out in

~/open-zwave-read-only

the patch is in
~/dev-domoticz/patches

unless OZW is checked out in
~/dev-domoticz/open-zwave-read-only

In the later, that will work

current builds now work correctly again in travis as the online patch now only includes the OZW 1125 patch

**Comment by emontnemery** [comment]
Ah, then I understand. Ok, I'll have another look

**Comment by emontnemery** [comment]
It's confirmed OZW is checked out in
~/dev-domoticz/open-zwave-read-only, so it should be fine to merge this PR.

Travis log with extra prints:
```
$ ls -l ..
total 4
drwxrwxr-x 32 travis travis 4096 Aug 12 16:20 domoticz
The command "ls -l .." exited with 0.
$ ls -l
total 101272
-rw-rw-r--  1 travis travis        77 Aug 12 16:19 appversion.default
-rw-rw-r--  1 travis travis      1400 Aug 12 16:19 appveyor.yml
drwxrwxr-x 10 travis travis      4096 Aug 12 16:20 boost_1_67_0
-rw-rw-r--  1 travis travis 103363944 Apr 11 14:47 boost_1_67_0.tar.gz
drwxrwxr-x  6 travis travis      4096 Aug 12 16:19 cereal
-rw-rw-r--  1 travis travis     28398 Aug 12 16:19 CMakeLists.txt
drwxrwxr-x 83 travis travis      4096 Aug 12 16:19 Config
-rw-rw-r--  1 travis travis      4427 Aug 12 16:19 domoticz.sh
drwxrwxr-x  4 travis travis      4096 Aug 12 16:19 dzVents
-rw-rw-r--  1 travis travis      3085 Aug 12 16:19 getgit.cmake
drwxrwxr-x 12 travis travis     12288 Aug 12 16:19 hardware
-rw-rw-r--  1 travis travis     87991 Aug 12 16:19 History.txt
drwxrwxr-x  2 travis travis      4096 Aug 12 16:19 httpclient
-rw-rw-r--  1 travis travis      5930 Aug 12 16:19 INSTALL.md
drwxrwxr-x  2 travis travis      4096 Aug 12 16:19 json
drwxrwxr-x  2 travis travis      4096 Aug 12 16:19 libusb
-rw-rw-r--  1 travis travis     35147 Aug 12 16:19 License.txt
drwxrwxr-x  4 travis travis      4096 Aug 12 16:19 lua
drwxrwxr-x  2 travis travis      4096 Aug 12 16:19 main
-rw-rw-r--  1 travis travis      2488 Aug 12 16:19 makebeta
-rw-rw-r--  1 travis travis      2346 Aug 12 16:19 makerelease
drwxrwxr-x  2 travis travis      4096 Aug 12 16:19 Manual
drwxrwxr-x  2 travis travis      4096 Aug 12 16:19 MQTT
drwxrwxr-x  4 travis travis      4096 Aug 12 16:19 msbuild
drwxrwxr-x  2 travis travis      4096 Aug 12 16:19 notifications
drwxrwxr-x 11 travis travis      4096 Aug 12 16:20 open-zwave-read-only
drwxrwxr-x  2 travis travis      4096 Aug 12 16:19 patches
drwxrwxr-x  4 travis travis      4096 Aug 12 16:19 plugins
drwxrwxr-x  2 travis travis      4096 Aug 12 16:19 push
-rw-rw-r--  1 travis travis      1466 Aug 12 16:19 README.md
drwxrwxr-x  8 travis travis      4096 Aug 12 16:19 scripts
-rw-rw-r--  1 travis travis      3414 Aug 12 16:19 server_cert.pem
drwxrwxr-x  2 travis travis      4096 Aug 12 16:19 smtpclient
drwxrwxr-x  2 travis travis      4096 Aug 12 16:19 sqlite
drwxrwxr-x  2 travis travis      4096 Aug 12 16:19 tcpserver
drwxrwxr-x  2 travis travis      4096 Aug 12 16:19 tinyxpath
drwxrwxr-x  3 travis travis      4096 Aug 12 16:19 tools
-rwxrwxr-x  1 travis travis       607 Aug 12 16:19 updatebeta
-rwxrwxr-x  1 travis travis       490 Aug 12 16:19 updatedomo
-rwxrwxr-x  1 travis travis       624 Aug 12 16:19 updaterelease
drwxrwxr-x  2 travis travis      4096 Aug 12 16:19 webserver
drwxrwxr-x 13 travis travis      4096 Aug 12 16:19 www
drwxrwxr-x  2 travis travis      4096 Aug 12 16:19 zip
drwxrwxr-x  3 travis travis      4096 Aug 12 16:19 zlib
```

**Comment by gizmocuz** [comment]
Thanks for the verification !
The patch file, now contains 100% the pure 1125 PR ? As i see some changes in the file

**Comment by emontnemery** [comment]
@gizmocuz Yeah, it's only PR 1125 in patches\domoticz-open-zwave.patch

**Comment by gizmocuz** [comment]
Thanks !!

**Pull request merged**